xbmc/network/upnp/UPnPInternal.cpp

```diff
@@ -1150,22 +1174,24 @@ bool GetResource(const PLT_MediaObject* entry, CFileItem& item)
   }
 
   // look for subtitles
-  unsigned subs = 0;
+  unsigned subIdx = 0;
+
   for(unsigned r = 0; r < entry->m_Resources.GetItemCount(); r++)
   {
     const PLT_MediaItemResource& res  = entry->m_Resources[r];
     const PLT_ProtocolInfo&      info = res.m_ProtocolInfo;
-    static const char* allowed[] = { "text/srt"
-      , "text/ssa"
-      , "text/sub"
-      , "text/idx" };
-    for(const char* const type : allowed)
+
+    for (const std::string& type : ALLOWED_REAL_SUBTITLE_FORMAT)
     {
-      if(info.Match(PLT_ProtocolInfo("*", "*", type, "*")))
+      if (type == (std::string)info.GetContentType().Split("/").GetLastItem()->GetChars())
```

Comment:
please avoid the c-style cast. The `std::string` does have an operator== with char* pointers (http://www.cplusplus.com/reference/string/string/operators/)